lib/index.js

```diff
@@ -82,7 +87,17 @@ function logFacesAppender(config) {
     Object.keys(event.context).forEach((key) => {
       lfsEvent[`p_${key}`] = event.context[key];
     });
-
+    if (config.eventLayout) {
+      lfsEvent = config.eventLayout(event, Object.assign({}, lfsEvent));
+      if (!lfsEvent) {
+        // no event object returned, consider the event ignored
+        return;
+      }
+    }
+    if (lfsEvent.m == null) {
```

Comment:
`lfsEvent.m = null` can be a valid value by `eventLayout()`. 
Should we match strictly to `undefined` instead? 
```suggestion
    if (lfsEvent.m === undefined) {
```

Reply:
`null` isn't a valid value for it from the typescript type I set up for `LogFacesEvent`/`LogFacesLayoutFunction` which is why I went with treating them the same way, since LogFaces will just send them as `"null"` or `"undefined"` rather than an empty string, which would be the way you'd specify "I explicitly don't want a message".
That being said, from my perspective I don't mind having it be a strict undefined check instead of a null/undefined check.

Reply:
I see that for `lfsEvent.m`, it is protected by the typescript to be either `string` or `undefined` (optional).

---

For those who are not using typescript (not protected by the typings), `lfsEvent.m` could be overridden via the newly exposed `eventLayout?: LogFacesLayoutFunction` to any other values, such as empty string or `null`. 

In the case of setting explicitly `lfsEvent.m = null`, that would mean that the user does not want a message. Yet the check for `if (lfsEvent.m == null)` would populate the `lfsEvent.m` with the default message, which is not what the user wants. Thus, I believe it will be good to have that strict `undefined` check.

Reply:
I see that for `lfsEvent.m`, it is protected by the typescript to be either `string` or `undefined` (optional).

I think this is due to `lfsEvent.m` being no longer populated at the beginning. Hence, there needs to be some logic that if `lfsEvent.m` is untouched by `eventLayout()`, it needs to populate itself.

For those who are not using typescript (not protected by the typings), `lfsEvent.m` could be overridden via the newly exposed `eventLayout?: LogFacesLayoutFunction` to any other values, such as empty string or `null`. 

In the case of setting explicitly `lfsEvent.m = null`, that would mean that the user does not want a message. Yet the check for `if (lfsEvent.m == null)` would populate the `lfsEvent.m` with the default message, which is not what the user wants. Thus, I believe it will be good to have that strict `undefined` check.

Then again, if the user explicit sets `lfsEvent = undefined`, do we consider that `lfsEvent.m` to be untouched?

**Should we remove the special fallback handling logic, populate `lfsEvent.m` in the beginning and let `eventLayout()` supersede accordingly? That will make things simpler. @ZachHaber, what do you think?**

Reply:
I wanted to make sure that `m` wasn't populated in the beginning due to the warning that `util.format` is slow in the node docs. So I decided that if the user populates the value, we don't need to calculate things.

It's pretty common in is world to default on null/undefined and treat them pretty synonymously. In my mind, `null` is a mistake for LogFaces as it ends up sending the word "null" to the server instead of an empty string.

The only option we have for `m` was _never_ defined is `"m" in lfEvent` which is doable, but i still feel it makes sense to replace null and undefined both due to their feeling like a mistake. `lfEvent.m ??= format(...)`

Reply:
Ah I see where you are coming from. If the user decides to overwrite `lfsEvent.m`, not having it in the first place would save the performance cost of the `util.format`.

And if the user did not touch `lfsEvent.m`, it will be populated after the `eventLayout()`.

If a user sets it explicitly to `null`, what do you suggest? Should `lfsEvent.m` be populated or should we send `null` over and let user discover the error?

Reply:
Personally, I tend to like using nullish coalescing for this sort of thing and just treat them the same. We can just write in the docs/jsdoc comment that if they want a blank message they need to populate it with an empty string rather than null or undefined as those will be populated normally.

Reply:
I agree. This kind of situation is a grey area and I'm glad to have heard your perspective. 

I'm OK with your suggestion. 🤗

Reply:
@ZachHaber This PR should be good to go once the above comments are addressed. 🤗